### ISSUE TITLE
glfwHideWindow() blocks certain events and window remains visible

### ISSUE THREAD

**Issue by NathanielGlover** [glfw/glfw#1300]
#### Sample Code:
```
#include <iostream>
#include <GLFW/glfw3.h>

void errorCallback(int, const char*);

int main() {
    glfwSetErrorCallback(errorCallback);
    glfwInit();
    GLFWwindow* win = glfwCreateWindow(400, 400, "Window", nullptr, nullptr);

    glfwShowWindow(win);
    while(!glfwWindowShouldClose(win)) {
        glfwPollEvents();
    }
    return 0;
}

void errorCallback(int error, const char* description) {
    std::cout << error << " " << description << std::endl;
}
```

#### Expected Behavior

The window is completely invisible and the program continues to run until it is told to quit from the dock (cmd + Q).

#### Actual Behavior

The window remains visible, and it no longer responds to resizing or clicks on the decoration bar (close, minimize, and fullscreen buttons). It is still possible to move the window around. No error output is generated.

Setting glfwWindowHint(GLFW_VISIBLE, GLFW_FALSE) before window creation has the same effect.

#### Environment Data

OS and version: macOS High Sierra 10.13.5
Release or commit: 3.2.1
Error messages: None
events output: I've spent several hours trying to compile and run it, but I'm not smart enough to get events.c working. If someone could supply me with a precompiled executable (for macOS), that would be perfect.

**Comment by elmindreda** [comment]
Cannot reproduce on macOS 10.13.5 or 10.13.6.

I can't see any call to `glfwHideWindow` above. How are you calling it and when?

**Comment by NathanielGlover** [comment]
Typo, my apologies. Should be there now.

**Comment by elmindreda** [comment]
I'm able to reproduce this with unbundled applications only.

**Comment by NathanielGlover** [comment]
Yes, after a full reinstall (directly from the git repo) I can confirm both of these things. I think my homebrew installation may have been a bit wonky (I've messed with it a few times in the past without reinstalling).
